### PR TITLE
add -fno-exceptions if cpp_eh=none is specified

### LINKED ISSUE #5347: cpp_eh=none should set -fno-exceptions on non-MSVC compilers
see title :)

### PR DESCRIPTION
**Pull request by ubsan**
Fixes #5347

### CODE REVIEW

**Comment by ubsan** [comment]
This should be done -- I ran all tests, and it looks like all failures are unrelated to the changeset.

File: docs/markdown/Builtin-options.md [hunk 1]
```diff
@@ -144,3 +144,9 @@ shell32, ole32, oleaut32, uuid, comdlg32, advapi32.
 c_args, cpp_args, c_link_args, and cpp_link_args only affect native builds,
 when cross compiling they will not be applied to binaries or libraries
 targeting the host system, only those being run on the build system.
+
+When using MSVC, `cpp_eh=default` will result in `/EHsc`, with the other values
+causing meson to pass `/EH[value]`, or nothing at all in the case of
+`cpp_eh=none`. When using other compilers, `cpp_eh=default` does nothing
+(allowing the default, exceptions, to work), while `cpp_eh=none` passes
+`-fno-exceptions`.
```
Comment (jpakkane) [inline docs/markdown/Builtin-options.md]: This should mention that this behaviour is only available starting with release 0.51.0.

Comment (ubsan) [reply docs/markdown/Builtin-options.md]: I'll do that later, currently have class.

**Comment by ubsan** [comment]
cc @jpakkane could this be merged?

**Comment by jon-turney** [comment]
Interestingly, this change breaks libui: 

`meson.build:6:0: ERROR: Value "sc" for combo option "cpp_eh" is not one of the choices. Possible choices are: "none", "default".`

(See https://travis-ci.org/jon-turney/meson-corpus-test/jobs/528532075)

beacuse that sets the default value of the cpp_eh option to 'sc' (whatver that means,  which I think is now only valid with MSVC, whereas previously it was ignored?)

Not sure if this needs fixing, but from the way this is documented, perhaps 'sc' should be accepted as an alias for 'default'?

**Comment by ubsan** [comment]
@jon-turney hmm, seems reasonable. I would probably want to warn if someone sets `cpp_eh = "sc"` for non-windows builds though?

**Pull request merged**